format/index.md

```diff
@@ -1004,125 +790,176 @@ Content-Type: application/vnd.api+json
     }
   }
 }
 ```
 
-In order to remove a to-one relationship, specify `null` as the value of the
-relationship.
+A server **MAY** reject an attempt to do a full replacement of a to-many
+relationship. In such a case, the server **MUST** reject the entire update,
+and return a `403 Forbidden` response.
 
-Alternatively, a to-one relationship **MAY** be accessible at its relationship
-URL (see above).
+> Note: Since full replacement may be a very dangerous operation, a server
+may choose to disallow it. A server may reject full replacement if it has
+not provided the client with the full list of associated objects, and does
+not want to allow deletion of records the client has not seen.
 
-A `PUT` request sent to the URL of a relationship **SHOULD** update the
-relationship. For example:
+#### Responses <a href="#crud-updating-responses" id="crud-updating-responses" class="headerlink"></a>
 
-```text
-PUT /articles/1/links/author
-Content-Type: application/vnd.api+json
-Accept: application/vnd.api+json
+##### 204 No Content <a href="#crud-updating-responses-204" id="crud-updating-responses-204" class="headerlink"></a>
 
-{
-  "people": "12"
-}
-```
+A server **MUST** return a `204 No Content` status code if an update is
+successful and the client's current attributes remain up to date.
 
-A `PUT` request **SHOULD** succeed regardless of whether a relationship is
-currently defined.
+##### 200 OK <a href="#crud-updating-responses-200" id="crud-updating-responses-200" class="headerlink"></a>
 
-A to-one relationship **MAY** alternatively be added by sending a `POST`
-request with an individual primary resource to the URL of the relationship.
-For example:
+If a server accepts an update but also changes the resource(s) in other ways
+than those specified by the request (for example, updating the `updatedAt`
+attribute or a computed `sha`), it **MUST** return a `200 OK` response.
 
-```text
-POST /articles/1/links/author
-Content-Type: application/vnd.api+json
-Accept: application/vnd.api+json
+The response document for a `200 OK` **MUST** include a representation of
+the updated resource(s) as if a `GET` request was made to the request URL.
 
-{
-  "people": "12"
-}
-```
+##### 403 Forbidden <a href="#crud-updating-relationship-responses-403" id="crud-updating-relationship-responses-403" class="headerlink"></a>
 
-A `POST` request should only succeed if no relationship is currently defined.
+A server **MUST** return `403 Forbidden` in response to an unsupported request
+to update a resource or relationship.
 
-A to-one relationship **MAY** be removed by sending a `DELETE` request to
-the URL of the relationship. For example:
+##### 404 Not Found <a href="#crud-updating-responses-404" id="crud-updating-responses-404" class="headerlink"></a>
 
-```text
-DELETE /articles/1/links/author
-```
+A server **MUST** return `404 Not Found` when processing a request to modify
+a resource that does not exist.
 
```

Comment:
I agree with @kevin-brown's [comment above](https://github.com/json-api/json-api/pull/341#discussion_r24473002) and the importance of the clause "and that URI is capable of being defined as a new resource by the requesting user agent".

The [PUT spec](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6) goes on to say:

> The recipient of the entity MUST NOT ignore any Content-* (e.g. Content-Range) headers that it does not understand or implement and MUST return a 501 (Not Implemented) response in such cases.

A client could reasonably expect a 501 if the server does not support the JSON API `Content-Type`. However, if the server does support the JSON API content type, then there is an understanding between client and server about this specification. Namely, in this scenario, the understanding is that PUT-as-create is not supported by the server.

So, having passed the `Content-*` negotiation, I believe that this spec is within its rights to recommend status codes that will be most clear and useful to clients, as long as they don't violate HTTP. And I believe that a 404 / 410 response is much more useful to clients than further overloading 501, which I think will only lead to confusion.